### DISCUSSION TITLE
3D Point cloud rendering?

### DISCUSSION THREAD

**Discussion by HagaiHargil** [bevyengine/bevy#2593] [Q&A]
Hi, is Bevy a good fit for three-dimensional point cloud rendering? I don't need any physics, just a render loop where I can specify the pixels to "turn on" in the next iteration, and potentially some post-processing capabilities. I couldn't find any example code, but perhaps if this is easily achievable I could contribute a simplified version of my code back.

I'm currently using `kiss3d` with some success, however it's no longer maintained and I've encountered a few bugs that I'm not exactly sure how to fix, since I'm far from an expert in this field. Other libraries are either too low-level for me right now, or don't seem to provide this functionality.

Thanks in advance.

**Comment by samcarey** [comment]
I also need this. I'm currently resorting to rendering little PBR cubes. I'd also like to be able to render 1-D lines in 3D space instead of trying to draw little cylinders.

**Comment by rezural** [comment]
I am working on this exact use case right now. I'll put up a like to the source code when done.

**Reply by rezural** [reply]
Although, I am rendering using spheres. I would prefer to do this in a shader, but  not there yet.

It works ok, for about up to 5000 or so points.

**Reply by carlbean** [reply]
Sounds great! Is there anything you can share from your work? :)

**Reply by rezural** [reply]
Hey,

It is (or will be) open source, but it is kind of hacky right now, and highly coupled with the program it is written in.

The program is https://github.com/rezural/mesh-ripper

I think the changes only exist locally, and I need to have a look at the changes again, and push what is there. IIRC it is only 70% done, but I will at least put up a branch.
